cpp/src/strings/regex/regcomp.cpp

```diff
@@ -188,6 +188,42 @@ class regex_parser {
   std::vector<Item> _items;
   bool _has_counted{false};
 
+  /**
+   * @brief Parses octal characters at the current expression position
+   * to return the represented character
+   *
+   * Reads up to 3 octal digits. The first digit should be passed
+   * in `in_chr`.
+   *
+   * @param in_chr The first character of the octal pattern
+   * @return The resulting character
+   */
+  char32_t handle_octal(char32_t in_chr)
```

Comment:
Is `in_chr` initially not just `*_expr_ptr`?

Reply:
It feels awkward that `handle_octal` and `handle_hex` have different signatures. Reading the code as a whole I see why, but it indicates that the APIs are super tailored to the exact places that they're being called. I was hoping to unify them further. I suppose in the other case making `handle_hex` take a parameter _just_ for consistency is also strange, because we know that we'll always need to read exactly two characters there, right?

Reply:
Right. 
In the hex case we read the `x` (in a certain state) that tells us the next 2 characters are hex (ignoring the `x`): `\xA2`
In the octal case we read a number (in a certain state) `0-7` but this number is part of computing the output: `\123`

Reply:
I'll defer to you here. It would be nice to align these better, but there doesn't appear to be a very easy way to do it so if you don't see a solution then I'm fine leaving it as is.